staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1421,11 +1423,15 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 			return "", err
 		}
 
+		// keyFunc is used by the storage decorators and there is no way to pass a context
```

Comment:
What does it mean "no way to pass a context" - aren't you passing contexts in the next lines? :)

Reply:
actually, I have just realized we should change all the layers to accept the context. That would simplify this PR and allow to reuse of some code. One sec :)